plugins/publisher.proto

```diff
@@ -297,6 +302,8 @@ message Record {
     // If provided, this should be an end-user readable string describing what triggered the publish.
     // This should only be set for real time published records, where the cause may be interesting.
     string cause = 4;
+
+    string correlation_id = 5;
```

Comment:
This should probably have a comment indicating it's only expected on records sent to `WriteStream`.